### PR TITLE
Updating Github actions to latest ones and making Ubuntu 20.04 LTS default

### PR DESCRIPTION
**Pull request by diogormendes**
Just updating github/actions to latest versions

### CODE REVIEW

**Comment by ariya** [comment]
Nice! Any particular benefits we want to get from the updated versions?

File: .github/workflows/auto-backport.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ on:
 jobs:
   pr_info:
     name: Check if the commit should be backported
-    runs-on: ubuntu-latest
+    runs-on: ubutnu-20.04
```
Comment (ariya) [inline .github/workflows/auto-backport.yml:12]: Typo

Comment (diogormendes) [reply .github/workflows/auto-backport.yml]: Thank you !!

**Comment by ariya** [comment]
Isn't `-latest` meant to be the most recent LTS anyway (which GitHub will use)?

Also note that https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners says that:

> Note: The -latest virtual environments are the latest stable images that GitHub provides, and might not be the most recent version of the operating system available from the operating system vendor.

which likely means that we should just trust GitHub and stick with `-latest`.

**Comment by diogormendes** [comment]
> Nice! Any particular benefits we want to get from the updated versions?

No. Just wanted to be updated with the latest changes. In the majority, they are updating the default runtime to node16 on the actions along with other minor updates. It is a draft still to run everything and make sure everything is working as expected.

**Comment by diogormendes** [comment]
> Isn't `-latest` meant to be the most recent LTS anyway (which GitHub will use)?
> 
> Also note that https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners says that:
> 
> > Note: The -latest virtual environments are the latest stable images that GitHub provides, and might not be the most recent version of the operating system available from the operating system vendor.
> 
> which likely means that we should just trust GitHub and stick with `-latest`.

Yes, but still want to stick with one that we know works instead of being at the risk of getting broke in the 22.04 lunch as LTS.

File: .github/workflows/uberjar.yml [hunk 2]
```diff
@@ -73,19 +73,19 @@ jobs:
       run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/} | sed 's/[^-._a-zA-Z0-9]/-/g')"
       id: extract_branch
     - name: Check out the code (Dockerfile needed)
-      uses: actions/checkout@v2
+      uses: actions/checkout@v3
     - name: Download uploaded artifacts to insert into container
-      uses: actions/download-artifact@v2
+      uses: actions/download-artifact@v3
       with:
-          name: metabase-${{ matrix.edition }}-uberjar
```
Comment (paoliniluis) [inline .github/workflows/uberjar.yml:80]: I'm assuming this was an unintended change right?

File: .github/workflows/uberjar.yml [hunk 3]
```diff
@@ -73,19 +73,19 @@ jobs:
...
-          path: bin/docker/
+        name: metabase-${{ matrix.edition }}-uberjar
+        path: bin/docker/
     - name: Move the ${{ matrix.edition }} uberjar to the context dir
       run: mv bin/docker/target/uberjar/metabase.jar bin/docker/.
     - name: Set up Docker Buildx
       id: buildx
       uses: docker/setup-buildx-action@v1
       with:
-          driver-opts: network=host
```
Comment (paoliniluis) [inline .github/workflows/uberjar.yml:88]: same as above

**Review by dragonsahead — APPROVED** [review]
I agree with these changes, it's always a good idea to keep our pipeline actions up to date even if they moved from an LTS version to another one.

Also pinning down the ubuntu version makes sense, we should test 22.04 when it launches to see if there's any issue

File: .github/workflows/uberjar.yml (hunk 3 above)
Comment (diogormendes) [reply .github/workflows/uberjar.yml:88]: Auto-Lint

**Comment by ariya** [comment]
> Also pinning down the ubuntu version makes sense, we should test 22.04 when it launches to see if there's any issue

If the past is indicative of the future, I don't think 22.04 will be immediately available anyway in any GitHub hosted runner. It usually take forever (best case, few months) until GitHub prepare a suitable runner that runs on the latest LTS.

**Pull request merged**